File: backend/app/schemas/complaint.py
from typing import Literal

from pydantic import BaseModel


class CreateComplaintRequest(BaseModel):
    title: str
    description: str
    latitude: float
    longitude: float
    ward_name: str


class UpdateComplaintStatusRequest(BaseModel):
    status: Literal[
        "reported",
        "verified",
        "assigned",
        "in_progress",
        "resolved",
        "closed"
    ]


class ComplaintResponse(BaseModel):
    id: str

    title: str
    description: str

    category: str
    ai_category: str

    priority: str
    severity_score: int

    recommended_department: str

    status: str

    class Config:
        from_attributes = True

File: frontend/types/complaint.ts
export interface CreateComplaintRequest {
  title: string;
  description: string;
  latitude: number;
  longitude: number;
  ward_name: string;
}

export interface Complaint {
  id: string;

  title: string;
  description: string;

  category: string;
  ai_category: string;

  priority: string;
  severity_score: number;

  recommended_department: string;

  status: string;

  ward_name?: string;

  latitude?: number;
  longitude?: number;

  created_at?: string;
  updated_at?: string;
}